src/main/java/io/appium/java_client/remote/MobileOptions.java

```diff
@@ -512,4 +518,14 @@ protected T amend(String optionName, Object value) {
         setCapability(optionName, value);
         return (T) this;
     }
+
+    @Override
+    public Map<String, Object> asMap() {
+        AcceptedW3CCapabilityKeys acceptedW3CCapabilityKeys = new AcceptedW3CCapabilityKeys();
+
+        return unmodifiableMap(super.asMap().entrySet().stream().collect(Collectors.toMap(
+                entry -> (acceptedW3CCapabilityKeys.test(entry.getKey()) || entry.getKey().startsWith(APPIUM_PREFIX)) ?
```

Comment:
I would move this handler into a separate method for better readability. Also, in Appium server code we don't add prefix if a capability name already contains `:` character, not just the `appium:` prefix. Should we consider to do the same here?